### DISCUSSION TITLE
some  questions

### DISCUSSION THREAD

**Discussion by mizuse-emiya** [rafael-fuente/N-body-numerical-simulation#3] [Q&A] [answered]
Dear rafael, I have some questions about the program. I want to know how to set the number of the particles and let the simulation auto-run without pulling the time line below the figure. I just mean how to make it like your examples of the video on youtube. Thanks a lot.

**Comment by rafael-fuente** [comment, marked as answer]
For setting a new particle just create the particle with `Particle()` constructor and add to the current list of particles `par` with append() method.

Particle() constructor takes two arguments, the first is the initial position of the particle and the second is the initial velocity.
```
#create the particles
par.append(Particle(Vec2(2,5),Vec2(0.5,0.5)))
par.append(Particle(Vec2(5,2),Vec2(0.5,0.2)))
par.append(Particle(Vec2(3,3),Vec2(0.1,0.5)))
par.append(Particle(Vec2(0.6,2.5),Vec2(0.5,0.5)))
```

For my youtube video I plot the simulation frame by frame and the I merged all frames in to a video using [ffmpeg](https://ffmpeg.org/)

An alternative for doing directly in matplotlib is using [matplotlib.animation.ArtistAnimation](https://matplotlib.org/gallery/animation/dynamic_image2.html), but unless you want make a video to post it somewhere for testing it's better an slider because it allows seeing easily any instant of the simulation.


For the frame by frame approach remplace the matplotlib code in the end of N-body simulation.py with this:

```
plt.style.use('dark_background')
fig = plt.figure(figsize=(7, 7))
ax = fig.add_subplot(1,1,1)

ax.axis('equal')
ax.axis([-1, 30, -1, 30])
ax.set_title('n =' + str(3))
ax.set_xlabel('x (t)')
ax.set_ylabel('y (t)')
ax.xaxis.set_visible(False)
ax.yaxis.set_visible(False)

circle = [None]*n
line  = [None]*n
for i in range(n):
    circle[i] = plt.Circle((par[i].solpos.x[0], par[i].solpos.y[0]), 0.08, ec="w", lw=2.5, zorder=20)
    ax.add_patch(circle[i])
    line[i] = ax.plot(par[i].solpos.x[:0],par[i].solpos.y[:0])[0]
    
fig.savefig('animation'+str(0)+'.png')

for i in range (1,steps+1):

    for j in range(n):
        circle[j].center = par[j].solpos.x[i], par[j].solpos.y[i]
        line[j].set_xdata(par[j].solpos.x[:i+1])
        line[j].set_ydata(par[j].solpos.y[:i+1])
    fig.savefig('animation'+str(i)+'.png')


```


Sorry for the delay in replying, I missed the notification.
